File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if(nums.size() == 1){
            return true;
        }
        int p=nums[0];
        int n = nums.size();
        for(int i=0;i<n;i++){
            p = max(p-1,nums[i]);
            if(p==0 and i!=n-1){
                return false;
            }
        }
        return true;
    }
};

File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let maxJump = nums[0];
    for(let i=0;i<nums.length;i++){
        maxJump = Math.max(maxJump-1,nums[i]);
        if(maxJump <= 0 && i!== nums.length){
            return false;
        }
    }
    return true;
};